### DISCUSSION TITLE
QuestPDF Outlines support

### DISCUSSION THREAD

**Discussion by lgsndbox** [QuestPDF/QuestPDF#181] [Q&A] [answered]
Hello,

Does QuestPDF support Outlines (often called bookmarks)?
I couldn't find a reference in the documentation.
Maybe simply creating Sections will create the corresponding Outlines? 
I did a quick test, and there was no /Outlines object in the generated PDF.

Thank you very much.

**Comment by MarcinZiabek** [comment, marked as answer]
Hello 😁 There is no explicit method in SkiaSharp API that could be used to draw outlines. However, there is [a generic method DrawAnnotation](https://docs.microsoft.com/en-us/dotnet/api/skiasharp.skcanvas.drawannotation?view=skiasharp-2.80.2) that expects binary data as an argument. Possibly it can be used to insert an outline. I haven't investigated much. Unfortunatelly, SkiaSharp mostly calls unmanaged Skia, so there is not much to find an inspiration from...

We can try to investigate it together 😁

**Reply by lgsndbox** [reply]
Hello,

Thanks, it does answer my question.

As for:

> We can try to investigate it together 😁

I wouldn't want you to waste your time:
- I have a very limited knowledge of PDF.
- I just discovered QuestPDF.
- I have never heard of Skia before.

So I'm not sure I would be of a great help. I looked into your suggestion, but I can't see how the [DrawAnnotation ](https://docs.microsoft.com/en-us/dotnet/api/skiasharp.skcanvas.drawannotation?view=skiasharp-2.80.2) method could help.
What mechanism could transform the Annotation's SKData instance into a PDF Outline? 
I don't even understand what an SKData really represent or is used for.

**Reply by MarcinZiabek** [reply]
Possibly SKData is a byte array that will be injected into PDF document. What form is required, there is not much documentation 😅

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
